torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -888,6 +888,10 @@ std::pair<NvrtcFunction, std::string> nvrtcCompile(
     int id,
     c10::optional<int> opt_block_size) {
   FUSER_PERF_SCOPE("executor_utils::NVRTC");
+  TORCH_CHECK(
+      !isDisabled(DisableOption::ArchCheck),
+      "NVFuser Compile: arch check disabled, cannot compile any kernel");
```

Comment:
nit: this may be just fine with a warning as disabling the arch check is opt-in. For example, when the disable option is used for debugging, I might want to see the real nvrtc error message for whatever debugging purpose.

Reply:
Changed this one to `TORCH_WARN` but still added `TORCH_CHECK` at the end of this function, just didn't want to purely rely on the toolkit behavior when we are almost certain the kernel should fail to compile.